lib/sidekiq/grouping/batch.rb

```diff
@@ -31,21 +31,28 @@ def chunk_size
       end
 
       def pluck
+        chunk = []
         if @redis.lock(@name)
-          @redis.pluck(@name, chunk_size).map { |value| JSON.parse(value) }
+          loop do
+            single_chunk = @redis.pluck(@name, chunk_size).map { |value| JSON.parse(value) }
+            chunk.push(*single_chunk)
+            break unless worker_class_options['batch_at_once']
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.